src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1593,7 +1416,7 @@ ConsumeQueryResult(PGconn *connection, bool failOnError, int64 *rows)
 	 */
 	while (true)
 	{
-		PGresult *result = PQgetResult(connection);
+		PGresult *result = PQgetResult(connection->pgConn);
```

Comment:
Since we're changing this, is it possible to call GetRemoteCommandResult here or would you rather do that as a separate PR?